diff --git a/src/library/grDevices/src/cairo/Makefile.win b/src/library/grDevices/src/cairo/Makefile.win
index b302550bbdf..f174c5658fe 100644
--- a/src/library/grDevices/src/cairo/Makefile.win
+++ b/src/library/grDevices/src/cairo/Makefile.win
@@ -19,7 +19,7 @@ endif
 PKG_CPPFLAGS = -I$(R_HOME)/src/include -I. -I.. -DHAVE_CONFIG_H \
   $(arch_DEFS) -I"$(CAIRO_HOME)/win$(WIN)"
 
-PKG_LIBS = -L"$(CAIRO_HOME)/win$(WIN)" -lcairo -lpixman-1 -lpng -lz \
+PKG_LIBS = -L"$(CAIRO_HOME)/win$(WIN)" -lcairo -lfontconfig -lfreetype -lpng -lpixman-1 -lexpat -lharfbuzz -lbz2 -lintl -lz \
   -lgdi32 -lmsimg32 -L.. -lgrDevices
 
 SOURCES = cairoBM.c
