diff --git a/src/library/grDevices/src/cairo/cairoBM.c b/src/library/grDevices/src/cairo/cairoBM.c
index b58a8d9e7ae..bd9c17343ac 100644
--- a/src/library/grDevices/src/cairo/cairoBM.c
+++ b/src/library/grDevices/src/cairo/cairoBM.c
@@ -122,7 +122,7 @@ BM_Open(pDevDesc dd, pX11Desc xd, int width, int height)
 #ifdef HAVE_CAIRO_SVG
     else if(xd->type == SVG) {
         snprintf(buf, PATH_MAX, xd->filename, xd->npages + 1);
-        xd->cs = cairo_svg_surface_create(R_ExpandFileName(buf),
+        xd->cs = cairo_svg_surface_create(cairo_filepath(buf),
                                           (double)xd->windowWidth,
                                           (double)xd->windowHeight);
         res = cairo_surface_status(xd->cs);
@@ -145,7 +145,7 @@ BM_Open(pDevDesc dd, pX11Desc xd, int width, int height)
 #ifdef HAVE_CAIRO_PDF
     else if(xd->type == PDF) {
         snprintf(buf, PATH_MAX, xd->filename, xd->npages + 1);
-        xd->cs = cairo_pdf_surface_create(R_ExpandFileName(buf),
+        xd->cs = cairo_pdf_surface_create(cairo_filepath(buf),
                                           (double)xd->windowWidth,
                                           (double)xd->windowHeight);
         res = cairo_surface_status(xd->cs);
@@ -167,7 +167,7 @@ BM_Open(pDevDesc dd, pX11Desc xd, int width, int height)
 #ifdef HAVE_CAIRO_PS
     else if(xd->type == PS) {
         snprintf(buf, PATH_MAX, xd->filename, xd->npages + 1);
-        xd->cs = cairo_ps_surface_create(R_ExpandFileName(buf),
+        xd->cs = cairo_ps_surface_create(cairo_filepath(buf),
                                          (double)xd->windowWidth,
                                          (double)xd->windowHeight);
         res = cairo_surface_status(xd->cs);
@@ -273,7 +273,7 @@ static void BM_NewPage(const pGEcontext gc, pDevDesc dd)
 		cairo_surface_destroy(xd->cs);
 		cairo_destroy(xd->cc);
                 snprintf(buf, PATH_MAX, xd->filename, xd->npages);
-                xd->cs = cairo_svg_surface_create(R_ExpandFileName(buf),
+                xd->cs = cairo_svg_surface_create(cairo_filepath(buf),
                                                   (double)xd->windowWidth,
                                                   (double)xd->windowHeight);
                 res = cairo_surface_status(xd->cs);
@@ -301,7 +301,7 @@ static void BM_NewPage(const pGEcontext gc, pDevDesc dd)
 		cairo_surface_destroy(xd->cs);
 		cairo_destroy(xd->cc);
                 snprintf(buf, PATH_MAX, xd->filename, xd->npages);
-                xd->cs = cairo_pdf_surface_create(R_ExpandFileName(buf),
+                xd->cs = cairo_pdf_surface_create(cairo_filepath(buf),
                                                   (double)xd->windowWidth,
                                                   (double)xd->windowHeight);
                 res = cairo_surface_status(xd->cs);
@@ -328,7 +328,7 @@ static void BM_NewPage(const pGEcontext gc, pDevDesc dd)
                 cairo_surface_destroy(xd->cs);
                 cairo_destroy(xd->cc);
                 snprintf(buf, PATH_MAX, xd->filename, xd->npages);
-                xd->cs = cairo_ps_surface_create(R_ExpandFileName(buf),
+                xd->cs = cairo_ps_surface_create(cairo_filepath(buf),
                                                  (double)xd->windowWidth,
                                                  (double)xd->windowHeight);
                 res = cairo_surface_status(xd->cs);
diff --git a/src/library/grDevices/src/cairo/cairoBM.h b/src/library/grDevices/src/cairo/cairoBM.h
index bdb1b09a7bf..36a0fafa620 100644
--- a/src/library/grDevices/src/cairo/cairoBM.h
+++ b/src/library/grDevices/src/cairo/cairoBM.h
@@ -54,6 +54,15 @@ typedef enum {
 #  include <cairo-ps.h>
 # endif
 
+/* As of version 1.15.10 Cairo requires all paths be in UTF-8
+ * See: https://cgit.freedesktop.org/cairo/commit/?id=84fc0ce91d1a57d20500f710abc0e17de82c67df
+ */
+#if defined(_WIN32) && (CAIRO_VERSION >= CAIRO_VERSION_ENCODE(1,15,10))
+#define cairo_filepath(x) Rf_translateCharUTF8(Rf_mkChar(R_ExpandFileName(x)))
+#else
+#define cairo_filepath(x) R_ExpandFileName(x)
+#endif
+
 typedef struct {
     /* Graphics Parameters */
     /* Local device copy so that we can detect */
