diff --git a/src/gnuwin32/installer/header1.iss b/src/gnuwin32/installer/header1.iss
index 52f11986747..9516db51c60 100644
--- a/src/gnuwin32/installer/header1.iss
+++ b/src/gnuwin32/installer/header1.iss
@@ -44,7 +44,7 @@ Name: ko; MessagesFile: "Korean.isl"
 #include "CustomMsg.iss"
 
 [Tasks]
-Name: "desktopicon"; Description: {cm:CreateDesktopIcon}; GroupDescription: {cm:AdditionalIcons}; MinVersion: 0,6.0
+Name: "desktopicon"; Description: {cm:CreateDesktopIcon}; GroupDescription: {cm:AdditionalIcons}; MinVersion: 0,6.0; Flags: unchecked
 Name: "quicklaunchicon"; Description: {cm:CreateQuickLaunchIcon}; GroupDescription: {cm:AdditionalIcons}; MinVersion: 0,6.0; Flags: unchecked 
 Name: "recordversion"; Description: {cm:recordversion}; GroupDescription: {cm:regentries}; MinVersion: 0,6.0
 Name: "associate"; Description: {cm:associate}; GroupDescription: {cm:regentries}; MinVersion: 0,6.0; Check: IsAdmin
